### PR TITLE
RSpec integration test

### PR DESCRIPTION
**Pull request by touredri**
- [ ] Fixed n+1 problem

Capture d’écran 2023-07-26 092313

- [ ] Added integration test for all views using Capybara
 - [ ] Tested interactions

### CODE REVIEW

File: spec/views/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    @comment1 = @post1.comments.create(text: 'Comment 1', author: @user)
+    @comment2 = @post1.comments.create(text: 'Comment 2', author: @user)
+    @comment3 = @post2.comments.create(text: 'Comment 3', author: @user)
+    visit user_posts_path(@user)
+  end
+
+  it 'displays the user profile picture' do
+    # Add your Capybara code here to test for the profile picture
+  end
+
```
Comment (DammyShittu) [inline spec/views/post_index_spec.rb]: - [ ] Kindly add the missing test for the user's profile picture.

File: spec/views/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  it 'displays the number of comments and likes a post has' do
+    expect(page).to have_content('Comments: 2') # Assuming 2 comments for post1
+    expect(page).to have_content('Likes: 0')   # Assuming 0 likes for all posts
+  end
+
+  it "redirects to a post show page when clicking on a post's see more" do
+    click_link 'See More', match: :first
+    expect(current_path).to eq(user_post_path(@user, @post3))
+  end
+end
```
Comment (DammyShittu) [inline spec/views/post_index_spec.rb]: - [ ] For the post index spec, kindly add a scenario for asserting we can see a pagination section when the posts are more than can fit the view.

  image

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @mercymugambi, @citec-47 and @touredri,

## Highlights
✔️ Passing tests

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @touredri @citec-47  and @mercymugambi , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which haven't been addressed yet.


 ## Highlights :fire:
- Previous reviewer's changes implemented :heavy_check_mark: 
- Linters well setup and passing :heavy_check_mark: 
- Professional README :heavy_check_mark: 

 ## Required Changes ♻️
- [ ] Great job implementing the previous reviewer's suggestions. :clap:  However, I noticed that after making the requested changes, some of your tests are currently failing. So kindly fix them so that all the tests that are failing can pass as they were before :wink: 
image

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @NahnahAJ in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by touredri** [comment]
This is not for this milestone ohh! 
rspec spec/views

**Review by Zuhaib042 — APPROVED** [review]
Hi @touredri 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Implemented suggested changes
✅ Written tests for views
✅ Fixed n+1 queries
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**